src/NSubstitute/Core/DefaultForType.cs

```diff
@@ -1,10 +1,28 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.Reflection;
 
 namespace NSubstitute.Core
 {
     public class DefaultForType : IDefaultForType
     {
+        private static readonly Dictionary<Type, object> DefaultPrimitiveValues = new Dictionary<Type, object>
```

Comment:
Would it be faster to inline this as a big switch/`if/else if` in `DefaultInstanceOfValueType`?

e.g.
```csharp
if (returnType == typeof(bool)) return false;
else if (returnType == typeof(char)) return '\0\;
else if /* ... */
else Activator.CreateInstance(returnType);
```

Reply:
Thanks for the feedback. Actually, I was also in doubt regarding this 🤔 

I analyzed a few big repositories and found that not all the built-in value types are equally popular. The most popular ones appeared to be `bool`, `int` and `long` and `double`. Therefore, I left optimization only for those cases and decided to handle all the rest types as it was before. That's a reasonable trade-off IMO.